### PR TITLE
[DAT-1380] Write serialize-deserialize test with values from full range

### PR DESCRIPTION
**Pull request by hb0**
This PR adds a serialize-deserialize test which then checks that the first and last sensor points still have the same values.

We had some serialize-deserialize tests on Android, and a deserialization test in the `serialization` library but there should be a serialize-deserialize test in `serialization` to catch errors in the offsetter or formatter early, which are not caught in the `OffsetterTest`.

The test generates random values within the valid value range defined in `measurement.protos` documentation:

```
final var maxAcceleration = 16.0; // m/s²
final var maxRotation = 2 * 34.906585; // rad/s
final var maxDirection = 4911.994; // µT
```

With a signum which is inverted between sensor points, to also catch corner cases.

### CODE REVIEW

File: libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java [hunk 1]
```diff
@@ -184,6 +204,188 @@ void test() throws IOException, InvalidLifecycleEvents, UnsupportedFileVersion {
...
+    /**
+     * This test generates random sensor values within the supported value range and deserializes the bytes to ensure
+     * the values did not change.
+     */
+    @DisplayName("Happy Path test for the serialization and deserialization of 3d points.")
+    @Test
+    void testSerializeDeserialize() throws IOException, InvalidLifecycleEvents {
+
+        // Arrange - Events: start, stop (1 track)
+        final var batches = 100;
```
Comment (hb0) [inline libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java:216]: This test can also be used as a sample implementation for others who want to use our serialization.

**Review by hb0 — COMMENTED** [review]
pre-reviewed

File: libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java [hunk 2]
```diff
@@ -184,6 +206,210 @@ void test() throws IOException, InvalidLifecycleEvents, UnsupportedFileVersion {
...
+        final var sensorPointsPerBatch = 100;
+        final var sensorPoints = batches * sensorPointsPerBatch;
+        final long measurementStart = 1_660_000_000_000L;
```
Comment (muthenberg) [inline libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java:221]: If you assign a long constant, this should become a long var even without the datatype, doesn't it?

**Review by muthenberg — CHANGES_REQUESTED** [review]
Two smaller comments. Please check.

File: libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java (hunk 2 above)
Comment (hb0) [reply libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java:221]: Was a typo, thanks :)

File: libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java [hunk 3]
```diff
@@ -184,6 +206,210 @@ void test() throws IOException, InvalidLifecycleEvents, UnsupportedFileVersion {
...
+        final var builder = MeasurementBytes.newBuilder()
+                .setFormatVersion(PERSISTENCE_FILE_FORMAT_VERSION)
+                .addAllEvents(events)
+                .setLocationRecords(locations)
+                .setAccelerationsBinary(ByteString.copyFrom(accelerationBytes))
+                .setRotationsBinary(ByteString.copyFrom(rotationBytes))
+                .setDirectionsBinary(ByteString.copyFrom(directionBytes));
+
+        // Act - Deserialize
+        final var measurementBytes = builder.build().toByteArray();
```
[the remark this answers is not in the document]
Comment (hb0) [reply libs/deserializer/src/test/java/de/cyface/deserializer/BinaryFormatDeserializerTest.java:327]: Right. Fixed it. Thanks.
